### PR TITLE
FEAT :sparkles: : Codebrick editing and better Documentation Viewing

### PR DESCRIPTION
**Pull request by kaushalvivek**
Added support for editing added codebricks, also added documentation in snippet-insert modal for easy reference.

---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [ ] Description above provides context of the change
- [ ] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [ ] Unit tests coverage for changes (not needed for documentation changes)
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions.
- [ ] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/www_rbac/views.py [hunk 1]
```diff
@@ -4611,6 +4612,41 @@ def save_snippet(self, filename):
...
+            if metadata['description'] != "":
+                try:
+                    with open(snippets_path.joinpath(*[snippet_folder, 'description.md']), 'w') as f:
+                        f.write(metadata['description'])
+                except Exception as e:
+                    print(e)
+            # edit code
+            try:
+                with open(snippets_path.joinpath(*[snippet_folder, 'snippet.py']), 'w') as f:
+                    f.write(new_snippet)
```
Comment (Anubhav2701) [inline airflow/www_rbac/views.py:4640]: can we reuse the code here from add snippet?

Comment (Anubhav2701) [reply airflow/www_rbac/views.py:4640]: also, can we not add `edit_snippet` but put it in `add_snippet`

**Review by Anubhav2701 — CHANGES_REQUESTED** [review]

File: airflow/www_rbac/templates/airflow/editdag.html [hunk 2]
```diff
@@ -359,24 +429,27 @@ <h4 class="modal-title">Replace placeholder values in codebrick.</h4>
...
+    }
+
+    else {
       // adding a new line at the end of snippet.
       replaceTextInEditor(text)
     }
-
-    // disable the button when snippet is inserted. The button will be enabled
-    // after the codebrick div is toggled again.
-    $(btn).addClass('disabled');
```
Comment (Anubhav2701) [inline airflow/www_rbac/templates/airflow/editdag.html]: Don't remove this ...?

Comment (kaushalvivek) [reply airflow/www_rbac/templates/airflow/editdag.html]: Was causing this behavior:
- if you choose to abandon an attempt to edit code/insert snippet and close the modal, the button is unclickable until you toggle codebricks again. 
Found it problematic as if I choose to abandon an edit attempt and decide to insert the codeblock I can't do it without toggling the codebrick again.

Comment (Anubhav2701) [reply airflow/www_rbac/templates/airflow/editdag.html]: shobhit wanted this to happen. Anyway we can comment this for now, don't remove this

Comment (kaushalvivek) [reply airflow/www_rbac/templates/airflow/editdag.html]: Okay, will comment it. Editing.

**Review by Anubhav2701 — APPROVED** [review]

**Pull request merged**